x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/rolemapping/PutRoleMappingRequest.java

```diff
@@ -77,10 +78,19 @@ public ActionRequestValidationException validate(boolean validateMetadata) {
             validationException = addValidationError("role-mapping rules are missing", validationException);
         }
         if (validateMetadata && MetadataUtils.containsReservedMetadata(metadata)) {
-            validationException = addValidationError(
-                "metadata keys may not start with [" + MetadataUtils.RESERVED_PREFIX + "]",
-                validationException
-            );
+            if (metadata.containsKey(READ_ONLY_ROLE_MAPPING_METADATA_FLAG)) {
+                validationException = addValidationError(
+                    "metadata contains ["
```

Comment:
My understanding Is this intended as a temporary fix until Kibana adds support for not allowing updates to mappings with `_read_only`? We should probably remove this special handling after that?  

If it is a temporary fix for UX purposes only, I think something like `"[" + name + "] is a read-only mapping and cannot be updated."` that way we don't expose how we determine it's a read-only mapping.